pkg/versions/compare.go

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
AKS Engine already uses https://github.com/blang/semver--could it be used here instead of new code?

Reply:
No, since that package is quite strictly semver whereas the versions we are comparing are calver.
https://play.golang.org/p/aClagUgV2Fk